File: auth-service/src/main/java/com/edp/auth/model/UserRegisterRequestDto.java
package com.edp.auth.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserRegisterRequestDto {
    private String firstName;
    private String lastName;
    private String username;
    private String email;
    private String password;
    private LocalDate birthdate;
    private String phoneNumber;
    private String department;
    private String position;
    private boolean admin;
    private Long reportsToId;
}

File: edp-frontend/src/app/auth/model/user-register-request.dto.ts
export interface UserRegisterRequestDto {
  firstName: string;
  lastName: string;
  username: string;
  email: string;
  password: string;
  birthdate: string; 
  phoneNumber?: string;
  department?: string;
  position?: string;
  admin: boolean;
  reportsToId?: number;
}
